cpp/common/src/codingstandards/cpp/Operator.qll

```diff
@@ -264,3 +267,51 @@ class UserOverloadedOperator extends Function {
     not this.isCompilerGenerated()
   }
 }
+
+/**
+ * A `std::basic_istream` class, or something that can be used
+ * as one. Based on the BasicOStreamClass.
+ */
+private class BasicIStreamClass extends Type {
+  BasicIStreamClass() {
+    this.(Class).getName().matches("basic\\_istream%")
+    or
+    this.getUnspecifiedType() instanceof BasicIStreamClass
+    or
+    this.(Class).getABaseClass() instanceof BasicIStreamClass
+    or
+    this.(ReferenceType).getBaseType() instanceof BasicIStreamClass
+  }
+}
+
+/** An implementation of a stream insertion operator. */
+class StreamInsertionOperator extends Function {
```

Comment:
same here, check if this should go in filestreams and if that can be simplified afterwards , to have the calls use the operators [here](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/standardlibrary/FileStreams.qll#L55)